temporalio/worker/_workflow_instance.py

```diff
@@ -1258,6 +1292,7 @@ def _register_task(
         *,
         name: Optional[str],
     ) -> None:
+        self._assert_not_read_only("create task")
```

Comment:
nit: "cancel workflow"?

Reply:
oh that's my bad, github UI made this look like the check for `_cancel_external_workflow`